python_modules/dagster/dagster/core/origin.py

```diff
@@ -1,25 +1,40 @@
 from collections import namedtuple
+from typing import List
 
 from dagster import check
 from dagster.core.code_pointer import CodePointer
 from dagster.serdes import create_snapshot_id, whitelist_for_serdes
+from dagster.utils import frozenlist
+
+DEFAULT_DAGSTER_ENTRY_POINT = frozenlist(["dagster"])
+
+
+def get_python_environment_entry_point(executable_path: str) -> List[str]:
+    return frozenlist([executable_path, "-m", "dagster"])
 
 
 @whitelist_for_serdes
 class RepositoryPythonOrigin(
-    namedtuple("_RepositoryPythonOrigin", "executable_path code_pointer container_image"),
+    namedtuple(
+        "_RepositoryPythonOrigin", "executable_path code_pointer container_image entry_point"
+    ),
 ):
     """
     Derived from the handle structure in the host process, this is the subset of information
     necessary to load a target RepositoryDefinition in a "user process" locally.
     """
 
-    def __new__(cls, executable_path, code_pointer, container_image=None):
+    def __new__(cls, executable_path, code_pointer, container_image=None, entry_point=None):
         return super(RepositoryPythonOrigin, cls).__new__(
             cls,
             check.str_param(executable_path, "executable_path"),
             check.inst_param(code_pointer, "code_pointer", CodePointer),
             check.opt_str_param(container_image, "container_image"),
+            (
+                frozenlist(check.list_param(entry_point, "entry_point", of_type=str))
+                if entry_point != None
+                else None
+            ),
```

Comment:
@alangenfeld how does the 'cast to frozenlist' part work if you move it to Namedtuple, I don't think you can override the constructor in those?

Reply:
you can still override new if you use the not-as-nice style of declaring

 example @ https://github.com/dagster-io/dagster/blame/master/python_modules/dagster/dagster/core/definitions/events.py#L37